pimd/pim_mroute.c

```diff
@@ -154,11 +154,14 @@ int pim_mroute_msg_nocache(int fd, struct interface *ifp, const kernmsg *msg)
 	sg.src = msg->msg_im_src;
 	sg.grp = msg->msg_im_dst;
 
-	if (!pim_ifp) {
-		if (PIM_DEBUG_MROUTE)
+
+	if ((!pim_ifp) || (!pim_ifp->pim_enable)) {
+		if (PIM_DEBUG_MROUTE_DETAIL)
 			zlog_debug(
-				"%s: PIM not enabled on interface, dropping packet to %pSG",
-				ifp->name, &sg);
+				"%s: Interface is not configured correctly to handle incoming packet, %s on interface %s",
```

Comment:
nit: IMO `interface` is kinda repetitive here. What about `"%s: Interface %s is not configured correctly (%s) to handle incoming packet"`? Also, showing a group is not a big deal here, or would be useful (as before this change)?

Reply:
Addressed.